contrib/devtools/github-merge.py

```diff
@@ -242,9 +236,13 @@ def main():
         subprocess.call([GIT,'branch','-q','-D',local_merge_branch],stderr=devnull)
 
     # Push the result.
-    reply = ask_prompt("Type 'push' to push the result to %s, branch %s." % (host_repo,branch))
-    if reply.lower() == 'push':
-        subprocess.check_call([GIT,'push',host_repo,'refs/heads/'+branch])
+    while True:
+        reply = ask_prompt("Type 'push' to push the result to %s, branch %s, or 'x' to exit without pushing." % (host_repo,branch)).lower()
```

Comment:
> May be nice to change this to be "type the remote and branch to confirm" so that you are forced to actually read where you're pushing it :)

Not sure this will help much. I don't think I've ever pushed something to the wrong place with this script after I made it take the target branch from the pull metadata on github.

> in a similar vein, may be good to also add an additional confirm if the branch is prefixed with a +, ie, a force push.

We don't support force-pushing from this script, and force pushes to master and all the version branches are disabled on github.